src/notification.c

```diff
@@ -433,6 +439,36 @@ parse_serialized_icon (GVariantBuilder  *builder,
       else if (xdp_validate_icon (sealed_icon, NULL, NULL))
         g_variant_builder_add (builder, "{sv}", "icon", icon);
     }
+  else if (strcmp (key, "file-descriptor") == 0)
+    {
+      g_autoptr(GError) local_error = NULL;
+      g_autoptr(XdpSealedFd) sealed_icon = NULL;
+
+      if (!check_value_type (key, value, G_VARIANT_TYPE_HANDLE, error))
+        return FALSE;
+
+      if (!G_IS_UNIX_FD_LIST (fd_list) || g_unix_fd_list_get_length (fd_list) == 0)
+        {
+          g_set_error_literal (error,
+                               XDG_DESKTOP_PORTAL_ERROR,
+                               XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                               "Invalid file descriptor: No Unix FD list given or empty");
+          return FALSE;
+        }
+
+      if (!(sealed_icon = xdp_sealed_fd_new_from_handle (value, fd_list, &local_error)))
+        {
+          g_warning ("Failed to seal fd: %s", local_error->message);
+          g_set_error_literal (error,
+                               XDG_DESKTOP_PORTAL_ERROR,
+                               XDG_DESKTOP_PORTAL_ERROR_INVALID_ARGUMENT,
+                               "Invalid file descriptor: The file descriptor needs to be sealable");
+          return FALSE;
+        }
+
+      if (xdp_validate_icon (sealed_icon, NULL, NULL))
+        g_variant_builder_add (builder, "{sv}", "icon", icon);
```

Comment:
yes, it's version 2 of the specs and the backend also needs to support them. not sure if we should add version property for the backend and for all new properties? @GeorgesStavracas what do you think?

Reply:
Portal backends may be versioned too, this is what we did with the Screenshot portal for reference. If it's important to preserve compatibility with older backends, then it might be worth versioning the org.freedesktop.impl.portal.Notification interface.

Reply:
gonna add the version property to `org.freedesktop.impl.portal.Notification` interface and convert the FD to a GFileIcon if the backend doesn't support the fd directly. At the same time if a backend supports v2 we can always give hand over an FD instead of the serialized byte icon.

Reply:
Sounds good to me